projects/plugins/jetpack/modules/masterbar/admin-menu/load.php

```diff
@@ -7,6 +7,35 @@
 
 namespace Automattic\Jetpack\Dashboard_Customizations;
 
+use Automattic\Jetpack\Status;
+
+require_once __DIR__ . '/class-dashboard-switcher.php';
+
+/**
+ * Check if the current request is an API request or not.
+ *
+ * @return bool
+ */
+function is_api_request() {
+	return ( defined( 'REST_REQUEST' ) && REST_REQUEST ) || 0 === strpos( $_SERVER['REQUEST_URI'], '/?rest_route=%2Fwpcom%2Fv2%2Fadmin-menu' );
+}
+
+/**
+ * Check if the current site is a domain only site.
+ *
+ * @return bool
+ */
+function is_domain_only_site() {
```

Comment:
Helper function extracted from the existing implementation so that we could reuse it in multiple places in this script.